app/geo/gui/src/ts/twxplore/gui/geo/attributeStrategies/classes/NumericFeatureAttribute.ts

```diff
@@ -17,36 +15,66 @@ export abstract class NumericFeatureAttribute implements FeatureAttribute {
     },
     addedFeature: MapFeature
   ): void {
-    updateNumericAttributeStateOfFeatureType(
-      attributeStatesOfFeatureType,
-      addedFeature,
-      this.name
-    );
-  }
-  buildInitialFeatureAttributeState(attributeStatesOfFeatureType: {
-    [featureAttributeName: string]: MapFeatureAttributeState;
-  }): void {
-    buildInitialFeatureAttributeStateNumeric(
-      attributeStatesOfFeatureType,
-      this.name
-    );
+    const attributeName = this.name;
+    const attributeKey = attributeName as keyof MapFeature;
+    //Specifying the attribute state of the addedFeature's FeatureType to be updated (e.g. the attribute state of 'frequency' of Transmissions)
+    let attributeStateOfAttributeOffFeatureType = attributeStatesOfFeatureType[
+      attributeName
+    ] as MapNumericFeatureAttributeState;
+    //If this is the first time coming across this attribute for the addedFeature's FeatureType
+    if (!attributeStateOfAttributeOffFeatureType) {
+      //Give the MapNumericAttributeState min/max the addedFeature's value for the attribute
+      attributeStatesOfFeatureType[attributeName] = {
+        range: {
+          min: addedFeature[attributeKey] as number,
```

Comment:
I see a bunch of duplicate expressions here, like addedFeature[attributeKey] as number.